Source/Atomic/UI/UIWidget.h

```diff
@@ -203,6 +204,17 @@ class ATOMIC_API UIWidget : public Object, public tb::TBWidgetDelegate
 
     void DeleteAllChildren();
 
+    /// searches for specified widget ID from the top of the widget tree, returns the 1st one found.
+    virtual UIWidget *FindWidget ( const String& searchid );
+    /// return all of the widgets of the specified classname and id that is not 0 from the current widget
+    virtual void SearchWidgetClass ( const String& className, UISelectItemSource *results );  
+    ///  return all of the widgets of the specified id and id that is not 0 from the current widget
+    virtual void SearchWidgetId ( const String& searchid, UISelectItemSource *results );
+    /// return all of the widgets with the specified text and id that is not 0 from the current widget
+    virtual void SearchWidgetText ( const String& searchText, UISelectItemSource *results );
+    /// print out the widget tree to stdout from the current widget
```

Comment:
Hm, I wonder if we should be using UISelectItemSource here as a general container?  It would probably be better to use a ```Vector<SharedPtr<UIWidget>>```, this will automatically be available to C#.  Automated JavaScript vector support is really not far off since it was added to C#.  In the meantime, on the JS side, it would take adding something like we do for getting scene children by name: https://github.com/AtomicGameEngine/AtomicGameEngine/blob/ab0157e4900011a0f70216e16f5de48fe9c43abe/Source/AtomicJS/Javascript/JSScene.cpp#L141 

What do ya think?